test-gensalt-extradata.c

```diff
@@ -149,7 +147,7 @@ main(void)
   char sref[6][CRYPT_GENSALT_OUTPUT_SIZE];
   char stst[CRYPT_GENSALT_OUTPUT_SIZE];
 
-  for (size_t i = 0; i < ARRAY_SIZE(testcases); i++)
+  for (int i = 0; i < (int) ARRAY_SIZE(testcases) - 1; i++)
```

Comment:
This also seems unnecessary...

Reply:
As commented above in a certain corner case we need a terminating element in the array.  With this change I make sure, we do not iterate the last element.